gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapAdapter.java

```diff
@@ -247,10 +257,21 @@ public LatLngBounds getViewport() {
   }
 
   @Override
-  public void renderTileOverlay() {
-    try (MapBoxOfflineTileProvider tileProvider =
-        new MapBoxOfflineTileProvider(context.getFilesDir())) {
-      map.addTileOverlay(new TileOverlayOptions().tileProvider(tileProvider));
-    }
+  public void renderTileOverlays(ImmutableSet<String> mbtilesFiles) {
+    stream(mbtilesFiles)
+        .forEach(
+            path -> {
+              File mbtiles = new File(context.getFilesDir(), path);
+              if (mbtiles.exists()) {
```

Comment:
Could you short circuit exit here instead and Timber.info the fact the the file was not found?

Reply:
57cb7dd9d31e68d8c97168a3778b7bcb62c173f7